contracts/cw20-base/src/msg.rs

```diff
@@ -107,6 +116,19 @@ pub enum ExecuteMsg {
     },
     /// Only with "approval" extension. Destroys tokens forever
     BurnFrom { owner: String, amount: Uint128 },
+    /// Only with the "marketing" extension. If authorised, updates marketing metadata.
```

Comment:
I think it is better solution, I was surprised those types are separated. However I see downside of this solution - you cannot update `cw20` without updating contract itself (as matches arm would not be fully covered). However it can be easly solved, just adding additional arm returning `"Err::Std("Not yet implemented")` or even just panicking with `unimplemented!()`.

Reply:
Yeah, for cw20-base, which should be a reference implementation, I think it does make sense to error if we don't implement everything (or implement anything out of spec). Only instantiation is custom.